jumanji/environments/routing/connector/generator.py

```diff
@@ -106,3 +107,54 @@ def __call__(self, key: chex.PRNGKey) -> State:
         step_count = jnp.array(0, jnp.int32)
 
         return State(key=key, grid=grid, step_count=step_count, agents=agents)
+
+
+class ParallelRandomWalkGenerator(Generator):
+    """Randomly generates `Connector` grids that may or may not be solvable. This generator places
+    start and target positions uniformly at random on the grid.
+    """
+
+    def __init__(self, grid_size: int, num_agents: int) -> None:
+        """Instantiates a `UniformRandomGenerator`.
+
+        Args:
+            grid_size: size of the square grid to generate.
+            num_agents: number of agents/paths on the grid.
+        """
+        super().__init__(grid_size, num_agents)
+        self.board_generator = ParallelRandomWalk(
+            self.grid_size, self.grid_size, self.num_agents
+        )
```

Comment:
It seems weird to me to define this `ParallelRandomWalk` class outside of this module and then just call its `generate_board` method. Could we have only one class that implements everything? As in, can we move everything from `ParallelRandomWalk` into this `ParallelRandomWalkGenerator` generator?

Reply:
I agree, moved all methods into generator